lib/ansible/modules/cloud/amazon/dms_replication_subnet_group.py

```diff
@@ -0,0 +1,248 @@
+            pass
+        elif paramname == 'SubnetIds':
+            subnets = []
+            for subnet in param_described.get('Subnets'):
+                subnets.append(subnet.get('SubnetIdentifier'))
+            for modulesubnet in modparams['SubnetIds']:
+                if modulesubnet in subnets:
+                    pass
+        else:
+            changed = True
+    return changed
+
+
+def create_replication_subnet_group(connection):
+    try:
+        params = create_module_params()
+        return replication_subnet_group_create(connection, **params)
+    except botocore.exceptions.ClientError as e:
+        module.fail_json(msg="Failed to create DMS replication subnet group.",
+                         exception=traceback.format_exc(),
+                         **camel_dict_to_snake_dict(e.response))
+    except botocore.exceptions.BotoCoreError as e:
+        module.fail_json(msg="Failed to create DMS replication subnet group.",
+                         exception=traceback.format_exc())
+
+
+def modify_replication_subnet_group(connection):
+    try:
+        modify_params = create_module_params()
+        return replication_subnet_group_modify(connection, **modify_params)
+    except botocore.exceptions.ClientError as e:
+        module.fail_json(msg="Failed to Modify the DMS replication subnet group.",
+                         exception=traceback.format_exc(),
+                         **camel_dict_to_snake_dict(e.response))
+    except botocore.exceptions.BotoCoreError as e:
+        module.fail_json(msg="Failed to Modify the DMS replication subnet group.",
+                         exception=traceback.format_exc())
+
+
+def delete_replication_subnet_group(connection):
+    return True
+
+
+def main():
+    argument_spec = dict(
+        state=dict(choices=['present', 'absent'], default='present'),
+        subnetgroupidentifier=dict(required=True),
+        subnetgroupdescription=dict(required=True),
+        subnetids=dict(type='list', required=True),
+    )
+    global module
+    module = AnsibleAWSModule(
+        argument_spec=argument_spec,
+        required_if=[],
+        supports_check_mode=False
+    )
+    exit_message = None
+    changed = False
+    if not HAS_BOTO3:
+        module.fail_json(msg='boto3 required for this module')
+
+    state = module.params.get('state')
+    aws_config_region, ec2_url, aws_connect_params = \
+        get_aws_connection_info(module, boto3=True)
+    dmsclient = get_dms_client(aws_connect_params, aws_config_region, ec2_url)
+    subnet_group = describe_subnet_group(dmsclient,
+                                         module.params.get('subnetgroupidentifier'))
+    if state == 'present':
+        if replication_subnet_exists(subnet_group):
+            if compare_params(subnet_group["ReplicationSubnetGroups"][0]):
+                changed = True
+                exit_message = modify_replication_subnet_group(dmsclient)
```

Comment:
why not implement check mode?

~~~python
if not module.check_mode:
    exit_message = modify_replication_subnet_group(dmsclient)
else: 
    exit_message = " a phrase that makes sense in check mode"
~~~

Reply:
also addressed